python/ray/tune/hpo_scheduler.py

```diff
@@ -0,0 +1,189 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import copy
+import numpy as np
+try:
+    import hyperopt as hpo
+except Exception as e:
+    hpo = None
+
+from ray.tune.trial import Trial
+from ray.tune.error import TuneError
+from ray.tune.trial_scheduler import TrialScheduler, FIFOScheduler
+from ray.tune.config_parser import make_parser
+from ray.tune.variant_generator import to_argv
+
+
+class HyperOptScheduler(FIFOScheduler):
+    """FIFOScheduler that uses HyperOpt to provide trial suggestions.
+
+    Requires HyperOpt to be installed. Uses the Tree of Parzen Estimators
+    algorithm. Note that this class takes in a loss attribute
+    rather than a reward.
```

Comment:
Should we change all of them to loss attribute?

Reply:
Hm I guess we could, but in separate PR

Reply:
I think we should still to `reward_attr` for consistency then (basically negate it in this pr). We can fix up / have a more easy to understand terminology in the followup.